services/src/main/java/org/exoplatform/webconferencing/WebConferencingService.java

```diff
@@ -0,0 +1,1235 @@
+
+  /** The settings service. */
+  protected final SettingService                     settingService;
+
+  /** The providers. */
+  protected final Map<String, CallProvider>          providers           = new ConcurrentHashMap<>();
+
+  /** The space service. */
+  protected SpaceService                             spaceService;
+
+  /** The user listeners. */
+  protected final Map<String, Set<UserCallListener>> userListeners       = new ConcurrentHashMap<>();
+
+  /**
+   * Instantiates a new web conferencing service.
+   *
+   * @param jcrService the jcr service
+   * @param sessionProviders the session providers
+   * @param hierarchyCreator the hierarchy creator
+   * @param organization the organization
+   * @param socialIdentityManager the social identity manager
+   * @param driveService the drive service
+   * @param listenerService the listener service
+   * @param settingService the setting service
+   */
+  public WebConferencingService(RepositoryService jcrService,
+                                SessionProviderService sessionProviders,
+                                NodeHierarchyCreator hierarchyCreator,
+                                OrganizationService organization,
+                                IdentityManager socialIdentityManager,
+                                ManageDriveService driveService,
+                                ListenerService listenerService,
+                                SettingService settingService) {
+    this.jcrService = jcrService;
+    this.sessionProviders = sessionProviders;
+    this.hierarchyCreator = hierarchyCreator;
+    this.organization = organization;
+    this.socialIdentityManager = socialIdentityManager;
+    this.driveService = driveService;
+    this.listenerService = listenerService;
+    this.settingService = settingService;
+  }
+
+  /**
+   * Gets the user info.
+   *
+   * @param id the id
+   * @return the user info
+   * @throws Exception the exception
+   */
+  public UserInfo getUserInfo(String id) throws Exception {
+    User user = organization.getUserHandler().findUserByName(id);
+    if (user != null) {
+      Identity userIdentity = socialIdentityManager.getOrCreateIdentity(OrganizationIdentityProvider.NAME, id, true);
+      if (userIdentity != null) {
+        Profile socialProfile = socialIdentityManager.getProfile(userIdentity);
+        @SuppressWarnings("unchecked")
+        List<Map<String, String>> ims = (List<Map<String, String>>) socialProfile.getProperty(Profile.CONTACT_IMS);
+        UserInfo info = new UserInfo(user.getUserName(), user.getFirstName(), user.getLastName());
+        if (ims != null) {
+          for (Map<String, String> m : ims) {
+            String imType = m.get("key");
+            String imId = m.get("value");
+            if (imId != null && imId.length() > 0) {
+              CallProvider provider = getProvider(imType);
+              if (provider != null && provider.isSupportedType(imType)) {
```

Comment:
It means that a provider could be registered for a given type (imType), but does not support this type ?

Reply:
At first look it is a double check. But getProvider() takes an one by a type from a map populated on the container instantiation (via plugin), using provider.getSupportedTypes(). Indeed, if some provider will decide to change supported types in runtime (it's possible), then this check will do the work. So far, we don't have such need or provider, but the possibility is here.

Reply:
ok